File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/delete/DeleteResponse.java
package fr.myprysm.vertx.elasticsearch.action.delete;

import fr.myprysm.vertx.elasticsearch.action.support.DocWriteResponse;
import fr.myprysm.vertx.elasticsearch.action.support.ShardId;
import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@DataObject
public class DeleteResponse extends DocWriteResponse {

    public DeleteResponse(JsonObject json) {
        super(json);
    }

    public DeleteResponse(DocWriteResponse other) {
        super(other);
    }

    public DeleteResponse(DeleteResponse other) {
        super(other);
    }

    public DeleteResponse(ShardId shardId, String index, String type, String id, long version, long seqNo, long primaryTerm, boolean forcedRefresh, org.elasticsearch.action.DocWriteResponse.Result result) {
        super(shardId, index, type, id, version, seqNo, primaryTerm, forcedRefresh, result);
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/delete/DeleteResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.delete

import fr.myprysm.vertx.elasticsearch.action.delete.DeleteResponse
import fr.myprysm.vertx.elasticsearch.action.support.ShardId
import fr.myprysm.vertx.elasticsearch.action.support.ShardInfo
import org.elasticsearch.action.DocWriteResponse.Result

fun DeleteResponse(
        forcedRefresh: Boolean? = null,
        id: String? = null,
        index: String? = null,
        primaryTerm: Long? = null,
        result: Result? = null,
        seqNo: Long? = null,
        shardId: fr.myprysm.vertx.elasticsearch.action.support.ShardId? = null,
        shardInfo: fr.myprysm.vertx.elasticsearch.action.support.ShardInfo? = null,
        type: String? = null,
        version: Long? = null): DeleteResponse = fr.myprysm.vertx.elasticsearch.action.delete.DeleteResponse().apply {

    if (forcedRefresh != null) {
        this.setForcedRefresh(forcedRefresh)
    }
    if (id != null) {
        this.setId(id)
    }
    if (index != null) {
        this.setIndex(index)
    }
    if (primaryTerm != null) {
        this.setPrimaryTerm(primaryTerm)
    }
    if (result != null) {
        this.setResult(result)
    }
    if (seqNo != null) {
        this.setSeqNo(seqNo)
    }
    if (shardId != null) {
        this.setShardId(shardId)
    }
    if (shardInfo != null) {
        this.setShardInfo(shardInfo)
    }
    if (type != null) {
        this.setType(type)
    }
    if (version != null) {
        this.setVersion(version)
    }
}

